src/boundaries.rs

```diff
@@ -16,204 +30,117 @@ pub mod boundary_type {
+        (BOTTOM_LEFT, HORIZONTAL_DOWN) => Some(CROSS),   // (└, ┬) => Some(┼)
+        (BOTTOM_LEFT, HORIZONTAL_UP) => Some(HORIZONTAL_UP), // (└, ┴) => Some(┴)
 
-        (boundary_type::HORIZONTAL_DOWN, boundary_type::HORIZONTAL_DOWN) => {
-            Some(boundary_type::HORIZONTAL_DOWN)
-        } // (┬, ┬) => Some(┬)
-        (boundary_type::HORIZONTAL_DOWN, boundary_type::HORIZONTAL_UP) => {
-            Some(boundary_type::CROSS)
-        } // (┬, ┴) => Some(┼)
-        (boundary_type::HORIZONTAL_DOWN, boundary_type::CROSS) => Some(boundary_type::CROSS), // (┬, ┼) => Some(┼)
+        (VERTICAL_LEFT, VERTICAL_LEFT) => Some(VERTICAL_LEFT), // (┤, ┤) => Some(┤)
+        (VERTICAL_LEFT, VERTICAL_RIGHT) => Some(CROSS),        // (┤, ├) => Some(┼)
+        (VERTICAL_LEFT, HORIZONTAL_DOWN) => Some(CROSS),       // (┤, ┬) => Some(┼)
+        (VERTICAL_LEFT, HORIZONTAL_UP) => Some(HORIZONTAL_UP), // (┤, ┴) => Some(┼)
 
-        (boundary_type::HORIZONTAL_UP, boundary_type::HORIZONTAL_UP) => {
-            Some(boundary_type::HORIZONTAL_UP)
-        } // (┴, ┴) => Some(┴)
-        (boundary_type::HORIZONTAL_UP, boundary_type::CROSS) => Some(boundary_type::CROSS), // (┴, ┼) => Some(┼)
+        (VERTICAL_RIGHT, VERTICAL_RIGHT) => Some(VERTICAL_RIGHT), // (├, ├) => Some(├)
+        (VERTICAL_RIGHT, HORIZONTAL_DOWN) => Some(CROSS),         // (├, ┬) => Some(┼)
+        (VERTICAL_RIGHT, HORIZONTAL_UP) => Some(CROSS),           // (├, ┴) => Some(┼)
 
-        (boundary_type::CROSS, boundary_type::CROSS) => Some(boundary_type::CROSS), // (┼, ┼) => Some(┼)
+        (HORIZONTAL_DOWN, HORIZONTAL_DOWN) => Some(HORIZONTAL_DOWN), // (┬, ┬) => Some(┬)
+        (HORIZONTAL_DOWN, HORIZONTAL_UP) => Some(CROSS),             // (┬, ┴) => Some(┼)
 
-        (_, _) => None,
-    }
-}
+        (HORIZONTAL_UP, HORIZONTAL_UP) => Some(HORIZONTAL_UP), // (┴, ┴) => Some(┴)
 
-fn find_next_symbol(first_symbol: &str, second_symbol: &str) -> Option<&'static str> {
-    if let Some(symbol) = combine_symbols(first_symbol, second_symbol) {
-        Some(symbol)
-    } else {
-        combine_symbols(second_symbol, first_symbol)
+        (a, b) => combine_symbols(b, a), // If we didn't find a match, swap them
     }
 }
 
-#[derive(PartialEq, Eq, Hash, Debug)]
-pub struct Coordinates {
+pub struct ScreenCanvas {
     x: usize,
     y: usize,
-}
-
-impl Coordinates {
-    pub fn new(x: usize, y: usize) -> Self {
-        Coordinates { x, y }
-    }
-}
-
-pub trait Rect {
-    fn x(&self) -> usize;
-    fn y(&self) -> usize;
-    fn rows(&self) -> usize;
-    fn columns(&self) -> usize;
-    fn right_boundary_x_coords(&self) -> usize {
-        self.x() + self.columns()
-    }
-    fn bottom_boundary_y_coords(&self) -> usize {
-        self.y() + self.rows()
-    }
-}
-
-pub struct Boundaries {
     columns: usize,
     rows: usize,
     boundary_characters: HashMap<Coordinates, BoundaryType>,
+    borders: HashMap<EdgeType, usize>,
```

Comment:
I think it should probably also have a `children: Box<Vec<ScreenCanvas>>` or similar too

Reply:
I like everything about this PR except this...
Most multiplexers keep data in this way, and while it's very comfortable for us developers it creates some issues for the users. For example, if we want to resize a pane outside of its parent. This can happen for example where:
* we have 4 rectangles on screen (by splitting horizontally and then splitting each one of the halves vertically)
* we resize the bottom-right pane up
* we then resize it back down to where it was
* we resize the same pane left

To my knowledge, we're the only multiplexer that supports this. Every other multiplexer would drag the pane's sibling with it in at least one of these directions. And I think it's because they keep a hierarchical map of the panes.